lbchaptic/src/main/kotlin/studio/lunabee/compose/foundation/haptic/LbcHapticFeedback.kt

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2024 Lunabee Studio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * LbcHapticFeedback.kt
+ * Lunabee Compose
+ *
+ * Created by Lunabee Studio / Date - 3/25/2024 - for the Lunabee Compose library.
+ */
+
+package studio.lunabee.compose.foundation.haptic
+
+import android.content.Context
+import android.os.Build
+import android.os.VibrationEffect
+import android.os.Vibrator
+import android.view.View
+import androidx.annotation.RequiresPermission
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.hapticfeedback.HapticFeedback
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalHapticFeedback
+import androidx.compose.ui.platform.LocalView
+import androidx.core.content.ContextCompat
+
+class LbcHapticFeedback(
+    val vibrator: Vibrator?,
+    val hapticFeedBack: HapticFeedback,
+    private val view: View,
+) {
+    /**
+     * Perform an haptic feedback. If it's not supported, [fallback] will be used. To avoid an infinite loop, [fallback] is also tested.
+     */
+    @RequiresPermission(value = "android.permission.VIBRATE")
+    fun perform(hapticEffect: LbcHapticEffect, fallback: LbcHapticEffect? = LbcHapticEffect.Compose.LongPress) {
+        if (vibrator == null) return
+        val supportedHapticEffect = getSupportedHapticEffect()
+        if (supportedHapticEffect.contains(hapticEffect)) {
+            when (hapticEffect) {
+                is LbcHapticEffect.Compose -> performCompose(hapticEffect = hapticEffect)
+                is LbcHapticEffect.CompositionPrimitives -> performCompositionPrimitives(hapticEffect)
+                is LbcHapticEffect.Predefined -> performPredefined(hapticEffect)
+                is LbcHapticEffect.FeedbackConstants -> performFeedbackConstants(view = view, hapticEffect = hapticEffect)
+            }
+        } else {
+            // Avoid infinite loop by testing the fallback availability.
+            fallback?.takeIf { supportedHapticEffect.contains(it) }?.let { perform(hapticEffect = fallback) }
+        }
```

Comment:
I don't think we need recursivity here

```suggestion
        (hapticEffect.takeIf { supportedHapticEffect.contains(it) } ?: fallback)?.let { effect ->
            when (effect) {
                is LbcHapticEffect.Compose -> performCompose(hapticEffect = effect)
                is LbcHapticEffect.CompositionPrimitives -> performCompositionPrimitives(hapticEffect = effect)
                is LbcHapticEffect.Predefined -> performPredefined(hapticEffect = effect)
                is LbcHapticEffect.FeedbackConstants -> performFeedbackConstants(view = view, hapticEffect = effect)
            }
        }
```